src/helper-x11/main.cpp

```diff
@@ -0,0 +1,97 @@
+/***************************************************************************
+* Copyright (c) 2021 Pier Luigi Fiorini <pierluigi.fiorini@gmail.com>
+* Copyright (c) 2013 Abdurrahman AVCI <abdurrahmanavci@gmail.com>
+*
+* This program is free software; you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation; either version 2 of the License, or
+* (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program; if not, write to the
+* Free Software Foundation, Inc.,
+* 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+***************************************************************************/
+
+#include <QCoreApplication>
+
+#include "Constants.h"
+#include "SignalHandler.h"
+#include "xorguserhelper.h"
+
+using namespace SDDM;
+
+int main(int argc, char *argv[])
```

Comment:
I don't like this nested tree of helpers upon helpers.
It works, but it seems unnecessarily convoluted.

I don't see why the existing helper app can't start fork and start X wait, then fork and start the user session.

Super worst case we move utmp logging from the helper app to the daemon, we can make the helper drop privileges earlier in the main code rather than after the fork which might make things even cleaner.

Reply:
The fact that it works probably is enough? I'd say if we want to do a refactoring, we should first get the wayland one in as well, which has a similar problem and do it all together rather than holding the features on implementation details.

Reply:
Do we need to land your utmp patch before? @davidedmundson

Reply:
I think they'll certainly make things easier and cleaner.  Means we only need to drop privileges once.

The utmp patches also require the display name to be sent to the daemon, so your addition above "for the future" is useful already! :)

Reply:
I made a rebase and adjusted it so the main sddm-helper ran XOrgUserHelper in process. 
https://github.com/davidedmundson/sddm/blob/user_xorg_after_utmp 

I have a few qDebugs and whatnot to remove, but the end result seems easier to follow.

Reply:
>Do we need to land your utmp patch before? @davidedmundson

If it's going to prove pragmatically easier, we could merge this first, and then I can do those rounds of simplification on top before we have a release.

Reply:
> I made a rebase and adjusted it so the main sddm-helper ran XOrgUserHelper in process.
> https://github.com/davidedmundson/sddm/blob/user_xorg_after_utmp
> 
> I have a few qDebugs and whatnot to remove, but the end result seems easier to follow.

404 error :(

Reply:
I will merge #1383 and rebase on top of it, then move the code to run the display server from sddm-helper-x11 to sddm-helper

Reply:
Maybe https://github.com/davidedmundson/sddm/tree/user_xorg_after_utmp works? Though it will be slightly out of date now maybe?

We can do it before or after, whichever is easier.